src/libraries/System.IO.Compression/src/System/IO/Compression/ZipHelper.cs

```diff
@@ -163,14 +163,18 @@ internal static void AdvanceToPosition(this Stream stream, long position)
         {
             long numBytesLeft = position - stream.Position;
             Debug.Assert(numBytesLeft >= 0);
-            while (numBytesLeft != 0)
+            if (numBytesLeft > 0)
             {
                 const int throwAwayBufferSize = 64;
-                int numBytesToSkip = (numBytesLeft > throwAwayBufferSize) ? throwAwayBufferSize : (int)numBytesLeft;
-                int numBytesActuallySkipped = stream.Read(new byte[throwAwayBufferSize], 0, numBytesToSkip);
-                if (numBytesActuallySkipped == 0)
-                    throw new IOException(SR.UnexpectedEndOfStream);
-                numBytesLeft -= numBytesActuallySkipped;
+                byte[] throwAwayBuffer = new byte[throwAwayBufferSize];
+                do
+                {
+                    int numBytesToSkip = (numBytesLeft > throwAwayBufferSize) ? throwAwayBufferSize : (int)numBytesLeft;
```

Comment:
Math.Min?

Reply:
This one is tricky because `numBytesLeft` is a `long`, so this would need to be:

```C#
int numBytesToSkip = (int)Math.Min(throwAwayBufferSize, numBytesLeft);
```

Meaning we would always need to cast. Thoughts?

Reply:
If this is almost never used, I'd go for the simpler code (which IMHO is Math.Min with a cast).